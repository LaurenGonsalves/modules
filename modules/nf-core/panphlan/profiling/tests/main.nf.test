nextflow_process {

    name "Test Process PANPHLAN_PROFILING"
    script "modules/local/panphlan/profiling.nf"
    process "PANPHLAN_PROFILING"

    test("Should run without failures") {

            setup {
                run("PANPHLAN_MAP") {
                    script "modules/local/panphlan/map.nf"
                    process {
                        """
                        input[0] = "Bacteroides_fragilis"
                        input[1] = "/workspace/biobakerymgx/modules/local/panphlan/test_data/Bacteroides_fragilis/Bacteroides_fragilis_pangenome.tsv" // replace w: PANPHLAN_DOWNLOADPANGENOME.out.pangenome
                        input[2] = Channel.of (
                            [
                                file("/workspace/biobakerymgx/modules/local/panphlan/test_data/Bacteroides_fragilis/") //replace w: PANPHLAN_DOWNLOADPANGENOME.out.indexes
                            ]
                        )
                        input[3] = Channel.of (
                            [
                                [id:"test1_1", single_end:true ],
                                file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz")
                            ]
                        )
                        """
                    }
                }
            }

        when
            process {
                """
                input[0] = PANPHLAN_MAP.out.mapping_file
                input[1] = "Bacteroides_fragilis"
                input[2] = "/workspace/biobakerymgx/modules/local/panphlan/test_data/Bacteroides_fragilis/Bacteroides_fragilis_pangenome.tsv" // replace w: PANPHLAN_DOWNLOADPANGENOME.out.pangenome
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}