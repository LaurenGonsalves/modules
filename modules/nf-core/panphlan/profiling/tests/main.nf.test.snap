{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "Bacteroides_fragilis_profile.tsv:md5,57201aadb842ec33e54aac93dcd548e2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,09cf69e7738a738a3ae5fbc93b024b95"
                ],
                "profile_matrix": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "Bacteroides_fragilis_profile.tsv:md5,57201aadb842ec33e54aac93dcd548e2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,09cf69e7738a738a3ae5fbc93b024b95"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-15T22:21:46.524551399"
    }
}